### ISSUE TITLE
if isset(var) generate assignation of tested var

### ISSUE THREAD

**Issue by rbaaboud** [smarty-php/smarty#319]
Hi,

Since Smarty 3.1.31, the `if isset(var)` syntax generate code like this : 

```php
<?php $_prefixVariable7=$_smarty_tpl->tpl_vars['someVar']->value[$_smarty_tpl->tpl_vars['varAssignedInSomeForeachLoop']->value->someGetterValue()];
if (isset($_prefixVariable7)) {?>
	<span class="someSpanSpecificClass"></span>
<?php }?>
```

So in this case, I have a PHP Notice that tells me `Undefined index: $_smarty_tpl->tpl_vars['varAssignedInSomeForeachLoop']->value->someGetterValue()`

I don't have this behavior in the 3.1.30 version.

**Comment by Visualq** [comment]
Same problem here

**Comment by uwetews** [comment]
How does the template source look?

**Comment by rbaaboud** [comment]
Hi,

Sorry for delay.
Best wishes for the new year :)

So, i give you another example, more simple I think.

in my .tpl file:

```
{if isset($reductionCollectionTree[$refCategory])}
    <span class="reduction"></span>
{/if}
```

Then, in compiled file:

```
<?php $_prefixVariable7=$_smarty_tpl->tpl_vars['reductionCollectionTree']->value[$_smarty_tpl->tpl_vars['refCategory']->value];
if (isset($_prefixVariable7)) {?>
    <span class="reduction"></span>
<?php }?>
```

Thanks for your support.

**Comment by jaymoulin** [comment]
Seems to be related with http://www.smarty.net/forums/viewtopic.php?t=26222. BTW, the fix should impact `$smarty` instead of all `isset` calls

**Comment by ruester** [comment]
There is a similar problem regarding isset calls with arrays:
`{if isset($var["foo"])} ... {/if}`
In version 3.1.29 this worked even if  `$var` was not set. The same behaviour has PHP itself.
With v3.1.31 or the current master branch this throws an error:
`Notice: Undefined index: $var`
Also see issue #347

**Comment by uwetews** [comment]
This was already fixed in the master branch with issue #347 
It will later be included in 3.1.32

**Comment by rbaaboud** [comment]
Thanks 👍
